Content.Client/VoiceMask/VoiceMaskNameChangeWindow.xaml.cs

```diff
@@ -28,54 +45,104 @@ public VoiceMaskNameChangeWindow()
 
         SpeechVerbSelector.OnItemSelected += args =>
         {
-            OnVerbChange?.Invoke((string?) args.Button.GetItemMetadata(args.Id));
+            OnVerbChange?.Invoke((string?)args.Button.GetItemMetadata(args.Id));
             SpeechVerbSelector.SelectId(args.Id);
         };
+
+        SpeechSoundSelector.OnItemSelected += args =>
+        {
+            OnSoundChange?.Invoke((string?)args.Button.GetItemMetadata(args.Id));
+            SpeechSoundSelector.SelectId(args.Id);
+        };
     }
 
-    public void ReloadVerbs(IPrototypeManager proto)
+    /// <summary>
+    ///     Loads the prototypes into their respective lists. DOES NOT make any UI changes, just making the lists.
+    /// </summary>
+    public void ReloadVerbsAndNoises(IPrototypeManager proto)
     {
+        // Before you ask, yes, this is currently the clearest way of doing this...
+        // You could do some spooky inheritance shenanigans with the prototypes but it would be kind of weird.
         foreach (var verb in proto.EnumeratePrototypes<SpeechVerbPrototype>())
         {
-            _verbs.Add((Loc.GetString(verb.Name), verb.ID));
+            _speechVerbs.Add((Loc.GetString(verb.Name), verb.ID));
         }
-        _verbs.Sort((a, b) => a.Item1.CompareTo(b.Item1));
+        _speechVerbs.Sort((a, b) => a.Item1.CompareTo(b.Item1));
+
+        foreach (var noise in proto.EnumeratePrototypes<SpeechSoundsPrototype>())
+        {
+            _speechSounds.Add((Loc.GetString(noise.Name), noise.ID));
+        }
+        _speechSounds.Sort((a, b) => a.Item1.CompareTo(b.Item1));
     }
 
-    public void AddVerbs()
+    /// <summary>
+    ///     Populates all the buttons with the values stored in the lists.
+    /// </summary>
+    public void AddVerbsAndSounds()
     {
-        SpeechVerbSelector.Clear();
+        PopulateOptionButton(SpeechVerbSelector, _selectedVerb, _speechVerbs);
+        PopulateOptionButton(SpeechSoundSelector, _selectedSound, _speechSounds);
+    }
 
-        AddVerb(Loc.GetString("chat-speech-verb-name-none"), null);
-        foreach (var (name, id) in _verbs)
-        {
-            AddVerb(name, id);
-        }
+    /// <summary>
+    ///     Updates the window to the current state. E.g make sure the shown name is the actual name being used etc..
+    /// </summary>
+    public void UpdateState(string selectedName, string? selectedVerb, string? selectedSound)
```

Comment:
Can't we just pass the BUI state directly into the window for it to handle?

Reply:
good idea!